File: leetCode/CSharp/28-implement-strStr.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        if (needle == null || haystack == null || needle.Length > haystack.Length) {
            return -1;
        }
        
        for (int i = 0; i < haystack.Length - needle.Length + 1; i++) {
            if (haystack.Substring(i, needle.Length) == needle) {
                return i;
            }
        }
        
        return -1;
    }
}

//TODO write test cases

File: leetCode/Python/28-Implement-strStr.py
# use slicing in python to compare with target string.
def strStr_slicing(haystack: str, needle: str) -> int:
    if haystack == None or needle == None:
        return -1
    
    for i in range(len(haystack) - len(needle) + 1):
        if haystack[i: i +len(needle)] == needle:
            return i
    return -1

#TODO implement strStr with different method


if __name__ == "__main__":
  result = strStr_slicing("", "")
  print(result)




